CMakeLists.txt

```diff
@@ -11,7 +11,8 @@ set(CMAKE_VERBOSE_MAKEFILE ON)
 macro(add_patch submodule patch working_directory patch_list)
   add_custom_command(OUTPUT ${patch}.applied
     WORKING_DIRECTORY ${working_directory}
-    COMMAND patch --forward -p0 < ${patchdir}/${submodule}/${patch} || true
+    COMMAND git reset --hard
```

Comment:
Why do we need this? This can discard all the local changes, which could be unexpected in some cases.

Reply:
Apologies, this is from a commit that was meant to be local to my repository -- I've found that deleting and recreating the build directory many times will reapply the patch. This is not necessary for upstream and has been removed in my latest push